plugins/com.google.cloud.tools.eclipse.appengine.newproject.test/src/com/google/cloud/tools/eclipse/appengine/newproject/CreateAppEngineStandardWtpProjectTest.java

```diff
@@ -72,14 +75,37 @@ public void testUnitTestCreated() throws InvocationTargetException, CoreExceptio
     assertJunitAndHamcrestAreOnClasspath();
   }
 
-  private void assertJunitAndHamcrestAreOnClasspath() throws CoreException, JavaModelException {
+  private void assertJunitAndHamcrestAreOnClasspath() throws CoreException {
     assertTrue(project.hasNature(JavaCore.NATURE_ID));
     IJavaProject javaProject = JavaCore.create(project);
     assertTrue(javaProject.findType("org.junit.Assert").exists());
     assertTrue(javaProject.findType("org.hamcrest.CoreMatchers").exists());
   }
 
   @Test
+  public void testAppEngineLibrariesAdded() throws InvocationTargetException, CoreException {
+    AppEngineStandardProjectConfig config = new AppEngineStandardProjectConfig();
+    config.setProject(project);
+    Library library = new Library("fake-library");
+    config.setAppEngineLibraries(Collections.singletonList(library));
+    CreateAppEngineStandardWtpProject creator = new CreateAppEngineStandardWtpProject(config, adaptable);
+    creator.execute(new NullProgressMonitor());
+    assertAppEngineContainerOnClasspath(library);
+  }
+
+  private void assertAppEngineContainerOnClasspath(Library library) throws CoreException {
+    assertTrue(project.hasNature(JavaCore.NATURE_ID));
+    IJavaProject javaProject = JavaCore.create(project);
+    boolean found = false;
+    for (IClasspathEntry iClasspathEntry : javaProject.getRawClasspath()) {
+      if (iClasspathEntry.getPath().equals(library.getContainerPath())) {
+        found = true;
```

Comment:
Option: you could simply return when you find it and fail at the end if you don't.